Source/KISAddonConfig.cs

```diff
@@ -22,52 +23,154 @@ sealed class KISAddonConfig : MonoBehaviour {
-        try {
-          var moduleInventory =
-            avPart.partPrefab.AddModule(typeof(ModuleKISInventory).Name) as ModuleKISInventory;
-          KIS_Shared.AwakePartModule(moduleInventory);
-          var baseFields = new BaseFieldList(moduleInventory);
-          baseFields.Load(nodeSettings.GetNode("EvaInventory"));
-          moduleInventory.podSeat = i;
-          moduleInventory.invType = ModuleKISInventory.InventoryType.Pod;
-          DebugEx.Fine("Pod inventory module(s) for seat {0} loaded successfully", i);
-        } catch {
-          DebugEx.Error("Pod inventory module(s) for seat {0} can't be loaded!", i);
+
+    IEnumerator LoadInventories ()
+    {
+      // Kerbal parts.
+      UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
+      UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
+
+      // Set inventory module for every pod with crew capacity.
+      DebugEx.Info("Loading pod inventories...");
+      for (loadedPartIndex = 0; loadedPartIndex < loadedPartCount; loadedPartIndex++) {
+        AvailablePart avPart = PartLoader.LoadedPartsList[loadedPartIndex];
+        if (!(avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
+              || avPart.name == RdKerbalEva
+              || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1)) {
+          if (HasPodInventories (avPart.partPrefab)) {
+            // pod inventories have already been added to the part by another
+            // mod, so skip the part (probably a dynamic crew capacity)
+			  DebugEx.Info("Skipping " + avPart.partPrefab);
+            continue;
+          }
+          DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
+          AddPodInventories (avPart.partPrefab, avPart.partPrefab.CrewCapacity);
         }
+        yield return null;
+      }
+      done = true;
+    }
+
+    public override bool IsReady ()
+    {
+      return done;
+    }
+
+    public override float ProgressFraction ()
+    {
+      return (float)loadedPartIndex / (float)loadedPartCount;
+    }
+
+    public override string ProgressTitle ()
+    {
+      return "Kerbal Inventory System";
+    }
+
+    public override void StartLoad ()
+    {
+      done = false;
+      loadedPartCount = PartLoader.LoadedPartsList.Count;
+      StartCoroutine (LoadInventories ());
+    }
+  }
+
+  public void Awake() {
+    List<LoadingSystem> list = LoadingScreen.Instance.loaders;
+    if (list != null) {
+      for (int i = 0; i < list.Count; i++) {
+        if (list[i] is KISConfigLoader) {
+          (list[i] as KISConfigLoader).owner = this;
+        }
+        if (list[i] is KISPodInventoryLoader) {
+          (list[i] as KISPodInventoryLoader).done = false;
+        }
+        if (list[i] is PartLoader) {
+          GameObject go = new GameObject();
+
+          var invLoader = go.AddComponent<KISPodInventoryLoader> ();
+          // Cause the pod inventory loader to run AFTER the part loader.
```

Comment:
How you know the ModuleManager has already did the job? Is it guaranteed for it to run *before* the part loader?

Reply:
Yes, ModuleManager is guaranteed to run before the part loader. I just checked KSP logs, and ModuleManager gets to run before any other mod's dll because it is in the top-level directory (there could be timing issues if there were other dlls in there)

Reply:
They way I know ModuleManager's loader runs first is I learned this method from ModuleManager itself.